lxd/storage/drivers/load_test.go

```diff
@@ -0,0 +1,18 @@
+package drivers
```

Comment:
IIRC that was the reasoning back then for having the `DefaultVMBlockFilesystemSize` func on the `common` struct so that there is a default defined which can then be overwritten by the PowerFlex driver.

Is there a specific reason for getting rid of this pattern?

Reply:
We can't load a driver from `lxd/project/limits` because we don't have easy access to a `state.State`. 

Initially I was going to grab an entire `Info{}` from the driver but that should really [require that the driver be loaded](https://github.com/canonical/lxd/blob/main/lxd/storage/drivers/driver_btrfs.go#L95). This was a compromise so that the default can be defined per-driver and also not require the drivers to be loaded.

The test is just designed to prevent this kind of behavior:
```go
func (d *btrfs) defaultVMBlockFilesystemSize() string {
	v := d.state.OS.whatever // state could be nil when used from DefaultVMBlockFilesystemSize
	// ...
}
```

Reply:
Ah sorry then what I mentioned was an initial idea we didn't end up implementing.

So you could have a `DefaultVMBlockFilesystemSize()` on the driver `common` struct which returns the default `100MiB` and the PowerFlex driver can overwrite it and return the `8GiB` instead. So same as you did in the PR but with exported funcs without adding it to the `Info{}` struct so it can be called directly.

IIRC the comment back then was that this would return driver specific configuration which is usually served via the `Info{}` struct.

Reply:
So is the problematic you are seeing that if you run `driverFunc().Info()` parts of the struct might not be set (like the Btrfs version you have linked)?

Reply:
And therefore you have added the unexported `defaultVMBlockFilesystemSize` so that you can access the value directly without going through the `Info{}` struct?

Reply:
Yes, that's precisely it. `Info{}` mixes stateless and stateful information so it's not easy to get the stateless stuff that way without hauling in a lot of context. What do you think about splitting `Info` into stateless (no `Load` required) and stateful (`Load` required) components rather than exposing a special function for each bit of stateless info? I'm not sure what to call those functions/structs off the top of my head (`Info`/`InfoLoaded`? :grimacing:).

Reply:
I'm happy with having this DefaultVMBlockFilesystemSize function that uses the unexported function on the driver. Perhaps we can change this later to a special "Stateless" info struct, but no need right now.